File: src/Entity/Utilisateur.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Security\Core\User\UserInterface;
use Symfony\Component\Serializer\Annotation\Groups;
use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;

#[ORM\Table(name: "utilisateur")]
#[ORM\Entity]
class Utilisateur implements UserInterface
{
    #[ORM\Column(name: "idUser", type: "integer", nullable: false)]
    #[ORM\Id]
    #[ORM\GeneratedValue(strategy: "IDENTITY")]
    private $iduser;

    #[ORM\Column(name: "nomUser", type: "string", length: 255, nullable: false)]
    private $nomuser;

    #[ORM\Column(name: "prenomUser", type: "string", length: 255, nullable: false)]
    private $prenomuser;

    #[ORM\Column(name: "emailUser", type: "string", length: 255, nullable: false)]
    private $emailuser;

    #[ORM\Column(name: "mdp", type: "string", length: 255, nullable: false)]
    private $mdp;

    #[ORM\Column(name: "nbPoints", type: "integer", nullable: false)]
    private $nbpoints;

    #[ORM\Column(name: "numTel", type: "integer", nullable: false)]
    private $numtel;

    #[ORM\Column(name: "role", type: "string", length: 255, nullable: false)]
    private $role;

    #[ORM\Column(name: "activationToken", type: "string", length: 250, nullable: true)]
    private $activationToken;

    #[ORM\Column(type: "string", length: 255, nullable: true)]
    private $faSecretKey;

    #[ORM\Column(type: "boolean")]
    private $is2faEnabled = false;

    #[ORM\Column(type: "boolean")]
    private $isDisabled = false;

    #[ORM\Column(name: "reclamation", type: "string", length: 255, nullable: true)]
    public $reclamation;
    #[ORM\Column(name: "image", type: "string", length: 255, nullable: true)]
    private $image;

    public function getIduser(): ?int
    {
        return $this->iduser;
    }

    public function getNomuser(): ?string
    {
        return $this->nomuser;
    }

    public function setNomuser(string $nomuser): static
    {
        $this->nomuser = $nomuser;

        return $this;
    }

    public function getPrenomuser(): ?string
    {
        return $this->prenomuser;
    }

    public function setPrenomuser(string $prenomuser): static
    {
        $this->prenomuser = $prenomuser;

        return $this;
    }

    public function getEmailuser(): ?string
    {
        return $this->emailuser;
    }

    public function setEmailuser(string $emailuser): static
    {
        $this->emailuser = $emailuser;

        return $this;
    }

    public function getMdp(): ?string
    {
        return $this->mdp;
    }

    public function setMdp(string $mdp): static
    {
        $this->mdp = $mdp;

        return $this;
    }

    public function getNbpoints(): ?int
    {
        return $this->nbpoints;
    }

    public function setNbpoints(int $nbpoints): static
    {
        $this->nbpoints = $nbpoints;

        return $this;
    }

    public function getNumtel(): ?int
    {
        return $this->numtel;
    }

    public function setNumtel(int $numtel): static
    {
        $this->numtel = $numtel;

        return $this;
    }

    public function getRoles(): ?string
    {
        // Transformez le champ $role en un tableau pour respecter l'interface UserInterface
        return $this->role;
    }

    public function setRole(string $role): static
    {
        $this->role = $role;

        return $this;
    }
    public function getReclamation(): ?string
    {
        return $this->reclamation;
    }

    public function setReclamation(?string $reclamation): self
    {
        $this->reclamation = $reclamation;

        return $this;
    }
    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(?string $image): self
    {
        $this->image = $image;

        return $this;
    }
    public function getActivationToken(): ?string
    {
        return $this->activationToken;
    }

    public function setActivationToken(?string $activationToken): self
    {
        $this->activationToken = $activationToken;

        return $this;
    }
  
    public function getUsername(): string
    {
        return $this->emailuser; // Assuming email is the username
    }
    public function getPassword(): string
    {
        return $this->mdp;
    }

  

    public function isIs2faEnabled(): ?bool
    {
        return $this->is2faEnabled;
    }
    public function getIsDisabled(): bool
    {
        return $this->isDisabled;
    }

    public function setIsDisabled(bool $isDisabled): void
    {
        $this->isDisabled = $isDisabled;
    }

    public function getfaSecretKey(): ?string
    {
        return $this->faSecretKey;
    }
    
    public function setfaSecretKey(?string $faSecretKey): self
    {
        $this->faSecretKey = $faSecretKey;
    
        return $this;
    }
    
    public function getIs2faEnabled(): bool
    {
        return $this->is2faEnabled;
    }
    
    public function setIs2faEnabled(bool $is2faEnabled): self
    {
        $this->is2faEnabled = $is2faEnabled;
    
        return $this;
    }
    public function getSalt()
    {
        // Salt is not needed with bcrypt
        return null;
    }

    public function eraseCredentials()
    {
       
    }
    public function setPassword(string $mdp): self
    {
        $this->mdp = $mdp;

        return $this;
    }
}

File: src/main/java/entities/Utilisateur.java
package entities;

public class Utilisateur {

    private int idUser;
    private String nomUser;
    private String prenomUser;
    private String emailUser;
    private String motDePasse;
    private int nbPoints;
    private int numeroTel;
    private int idDons; // L'identifiant du don associé à cet utilisateur

    private String role ; // "administrateur" ou "client"

    public Utilisateur(String nomUser, String prenomUser, String emailUser, int numTel) {
        this.nomUser=nomUser;
        this.prenomUser=prenomUser;
        this.emailUser=emailUser;
        this.numeroTel=numTel;
    }

    public Utilisateur(int id, String nom, String prenom, String email) {
        this.idUser=id;
        this.nomUser=nom;
        this.prenomUser=prenom;
        this.emailUser=email;
    }

    public String getEmailUser() {
        return emailUser;
    }

    public void setEmailUser(String emailUser) {
        this.emailUser = emailUser;
    }


    public Utilisateur (String nomUserTFText, String text, String id , int nbPoints, String nomUser , String prenomUser){}




    public Utilisateur(String nomUser, String prenomUser, String emailUser, String motDePasse,
                       int nbPoints, int numeroTel, String role) {
        this.nomUser = nomUser;
        this.prenomUser = prenomUser;
        this.emailUser = emailUser;
        this.motDePasse = motDePasse;
        this.nbPoints = nbPoints;
        this.numeroTel = numeroTel;
        this.role = role;
    }

    public Utilisateur(int idUser, String nomUser, String prenomUser, String emailUser, String motDePasse, int nbPoints, int numeroTel, String role) {
        this.idUser = idUser;
        this.nomUser = nomUser;
        this.prenomUser = prenomUser;
        this.emailUser = emailUser;
        this.motDePasse = motDePasse;
        this.nbPoints = nbPoints;
        this.numeroTel = numeroTel;
        this.role = role;
    }

    public Utilisateur() {
        this.idUser = idUser;
        this.nomUser = nomUser;
        this.prenomUser = prenomUser;
        this.emailUser = emailUser;
        this.motDePasse = motDePasse;
        this.nbPoints = nbPoints;
        this.numeroTel = numeroTel;
        this.role = role;
    }

    public int getIdUser() {
        return idUser;
    }

    public void setIdUser(int idUser) {
        this.idUser = idUser;
    }
    public String getNomUser() {
        return nomUser;
    }

    public void setNomUser(String nomUser) {
        this.nomUser = nomUser;
    }
    public String getPrenomUser() {
        return prenomUser;
    }

    public void setPrenomUser(String prenomUser) {
        this.prenomUser = prenomUser;
    }


    public String getMotDePasse() {
        return motDePasse;
    }

    public void setMotDePasse(String motDePasse) {
        this.motDePasse = motDePasse;
    }

    public int getNbPoints() {
        return nbPoints;
    }

    public void setNbPoints(int nbPoints) {
        this.nbPoints = nbPoints;
    }
    public int getNumeroTel() {
        return numeroTel;
    }

    public void setNumeroTel(int numeroTel) {
        this.numeroTel = numeroTel;
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }
    @Override
    public String toString() {
        return "utilisateur{" +
                "idUser=" + idUser +
                ", nomUser='" + nomUser + '\'' +
                ", prenomUser='" + prenomUser + '\'' +
                ", emailUser='" + emailUser + '\'' +
                ", motDePasse='" + motDePasse + '\'' +
                ", nbPoints=" + nbPoints +
                ", numeroTel='" + numeroTel + '\'' +
                ", role='" + role + '\'' +
                '}';
    }

    public Utilisateur(int idUser, int nbPoints, String nomUser, String prenomUser, String emailUser,int numTel) {
        this.idUser = idUser;
        this.nbPoints = nbPoints;
        this.nomUser = nomUser;
        this.prenomUser = prenomUser;
        this.emailUser = emailUser;
        this.numeroTel = numTel;
    }

    public Utilisateur(int nbPoints, String nomUser, String prenomUser, String emailUser, int numTel) {
        this.nbPoints = nbPoints;
        this.nomUser = nomUser;
        this.prenomUser = prenomUser;
        this.emailUser = emailUser;
        this.numeroTel = numTel;
    }

    public Utilisateur(int idUser, String nom, String email, int nbPoints, int idDons) {
        this.idUser = idUser;
        this.nomUser = nom;
        this.emailUser = email;
        this.nbPoints = nbPoints;
        this.idDons = idDons; // Assurez-vous d'initialiser idDons avec la valeur fournie
    }
    public int getIdDons() {
        return idDons;
    }

}
